File: leetcode code/259-3sum-smaller/259-3sum-smaller.java
class Solution {
    public int threeSumSmaller(int[] nums, int target) {
        Arrays.sort(nums);
        int res = 0;
        for(int i=0;i<nums.length;i++){
            int l = i + 1;
            int r = nums.length - 1;
            
            while(l<r){
                int temp = nums[i] + nums[l] + nums[r];
                if (temp<target){
                    res = res + (r-l);
                    l++;
                }
                else{
                    r--;
                }
            }
        }
        
        return res;
    }
}

File: leetcode code/259-3sum-smaller/259-3sum-smaller.py
class Solution:
    def threeSumSmaller(self, nums: List[int], target: int) -> int:
        nums.sort()
        res = 0
        
        for i in range(len(nums)):
            l = i + 1
            r = len(nums) - 1
            
            while l<r:
                temp = nums[i] + nums[l] + nums[r]
                
                if temp < target:
                    res += (r-l)
                    l += 1
                else:
                    r -= 1
        
        return res